File: src/shopperior-ng/src/app/core/data/list/models/shopping-list-model.ts
import { CategoryModel } from "../../category/models/category-model";
import { ListItemModel } from "./list-tem-model";
import { UserListPermissionModel } from "./user-list-permission-model";

export interface ShoppingListModel {
  guid: string;
  name: string;
  permissions: UserListPermissionModel[];
  categories: CategoryModel[];
  items: ListItemModel[];
}

File: src/Shopperior.Api/Shopperior.Application/ShoppingLists/Models/ShoppingListModel.cs
using Shopperior.Domain.Contracts.Categories.Models;
using Shopperior.Domain.Contracts.ListItems.Models;
using Shopperior.Domain.Contracts.ShoppingLists.Models;

namespace Shopperior.Application.ShoppingLists.Models;

public class ShoppingListModel : IShoppingListModel
{
    public Guid Guid { get; set; }
    public string Name { get; set; }
    public IEnumerable<IUserListPermissionModel> Permissions { get; set; }
    public IEnumerable<ICategoryModel> Categories { get; set; }
    public IEnumerable<IListItemModel> Items { get; set; }
}